creator-node/src/diskManager.js

```diff
@@ -0,0 +1,107 @@
+const path = require('path')
+const fs = require('fs')
+const config = require('./config')
+
+// use this set to cache existing directory paths we know we've created so we don't make extraneous file system calls
+let EXISTING_PATHS = new Set()
+
+// regex to check if a directory or just a regular file
+// if directory - will have both outer and inner properties in match.groups
+// else - will have just outer property, no inner
+const CID_DIRECTORY_REGEX = /\/(?<outer>Qm[a-zA-Z0-9]{44})\/?(?<inner>Qm[a-zA-Z0-9]{44})?/
+
+class DiskManager {
+  /**
+   * Return the storagePath from the config
+   */
+  static getConfigStoragePath () {
+    return config.get('storagePath')
+  }
+
+  /**
+   * Construct the path to a file or directory
+   *
+   * eg. if you have a file `Qmabcxyz`, use this function to get the path /file_storage/files/cxy/Qmabcxyz
+   * eg. if you have a dir `Qmdir123`, use this function to get the path /file_storage/files/r12/Qmdir123/
+   * Use `computeFilePathInDir` if you want to get the path for a file inside a directory.
+   *
+   * @dev Returns a path with the three characters before the last character
+   *      eg QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3GrHmuU6 will be eg /file_storage/muU/QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3GrHmuU6
+   * @param {String} fsDest file system destination, either filename or directory name
+   */
+  static computeBasePath (fsDest) {
```

Comment:
is this the 'base' of the full path or the path itself?

Reply:
not sure i follow this question

Reply:
is the value returned out of this function further modified or is it the entire path?

Reply:
renamed to computeFilePath